### PR TITLE
Trying to replicate same User Experience from official spotify website(adding Upgrade button for the Users with free account)

### PR DESCRIPTION
**Pull request by SV9045**
### Picture from Spotify Website.
Screen Shot 2021-07-07 at 12 27 37 AM

* removing unused import of NzModalService from auth.store.ts
* Changing some html structure of user-dropdown.component.html to accommodate button

### CODE REVIEW

**Review by trungk18 — CHANGES_REQUESTED** [review]
Thanks @SV9045 for the PR. I have two comments

1. For some reason, the product didn't get emitted. I am a premium user and the button is still displaying for me. Could you please take a look?

2. The Upgrade button height is larger than the other button height. 36px vs 32px. I would like to see them in equal height.

Screenshot 2021-07-07 at 10 14 52 PM

**Comment by SV9045** [comment]
Hi @trungk18,

Thank you for your comments, I will take a look and update soon.

Thanks!

**Comment by SV9045** [comment]
@trungk18 , I made changes as per comment. It was silly mistake really.

Screen Shot 2021-07-07 at 10 39 14 AM

Thank you for your support.

**Comment by trungk18** [comment]
@SV9045 Thanks, the button size looks good. There is still one issue. 

When I am loading the page, the Upgrade button display and then disappear. You probably need to do an additional check for if the value is truthy. 


https://user-images.githubusercontent.com/6767322/124946884-f69e5600-e041-11eb-9faf-771199e3fdb2.mp4

File: libs/web/shell/ui/user-dropdown/src/lib/user-dropdown.store.ts [hunk 1]
```diff
@@ -1,16 +1,18 @@
-import { ComponentStore } from '@ngrx/component-store';
 import { AuthStore } from '@angular-spotify/web/auth/data-access';
-import { Observable } from 'rxjs';
 import { Injectable } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { Observable } from 'rxjs';
 
 @Injectable()
 export class UserDropdownStore extends ComponentStore<Record<string, string>> {
```
Comment (trungk18) [inline libs/web/shell/ui/user-dropdown/src/lib/user-dropdown.store.ts:7]: I don't think think we need a store for this component, can we just inject the `AuthStore` directly into the component and take the value from there?

**Comment by SV9045** [comment]
Hi @trungk18,

Sure, will take a look. I have not experienced this one.

File: libs/web/shell/ui/user-dropdown/src/lib/user-dropdown.store.ts (hunk 1 above)
Comment (SV9045) [reply libs/web/shell/ui/user-dropdown/src/lib/user-dropdown.store.ts:7]: Hi @trungk18,

Yes we can, but I just followed your steps as you brought userAvatar and userName from AuthStore to here and then inject this store to component.

I will modify this too. Also Do you want me to bring userName and userAvatar directly from AuthStore to component ?

**Comment by SV9045** [comment]
Hi @trungk18 , 
I removed user-dropdown.store.ts and connect all user related variables to AuthStore directly.
I'm attaching video, please have a look.

https://user-images.githubusercontent.com/68339306/125143433-b645e300-e0e8-11eb-9594-1051800dad92.mov

**Pull request merged**